### PR TITLE
Enable debug symbols for binaries when make DEBUG=1

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Setting DEBUG=1 during build will now result in a binary containing the
debug symbols. The same applies to the nix build when setting "DEBUG=1"
in `nix/default.nix`.

#### Which issue(s) this PR fixes:

Fixes https://github.com/cri-o/cri-o/issues/3528
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM once the tests are happy

File: nix/default.nix [hunk 1]
```diff
@@ -38,6 +38,7 @@ let
       buildInputs = old.buildInputs ++ (with pkgs; [ systemd ]);
       src = ./..;
       EXTRA_LDFLAGS = ''-linkmode external -extldflags "-static -lm"'';
+      # DEBUG = 1; # to enable debug symbols inside the binary
```
Comment (haircommander) [inline nix/default.nix:41]: what's the purpose of this comment? describing what needs to be done to set debugging symbols?

Comment (saschagrunert) [reply nix/default.nix:41]: Yes exactly. :)

Comment (haircommander) [reply nix/default.nix:41]: I would suggest the rewrite 
```
# DEBUG = 1; # Uncomment this line to enable debug symbols in the binary 
```
but it's a nit. I'm fine with a follow up if you want

Comment (saschagrunert) [reply nix/default.nix:41]: Fixed as suggested :+1:

**Comment by saschagrunert** [comment]
/test kata-containers
connection timeout to azure

**Pull request merged**